### ISSUE TITLE
[Bug?] Cluster DNS on Windows

### ISSUE THREAD

**Issue by skinny** [Azure/acs-engine#418]
Hi,

Not sure this is even an ACS issue but it seems the best place to create this issue. When I deploy a cluster with the ACS engine all works fine. However it seems that windows pods end up with two network interfaces (inside the container) 

If I understand correctly the 172.x one is the default docker NAT interface and the other one is the kubernetes transparent network. Is this even correct ? 

Both of them get the 10.3.0.10 (my cluster DNS) but sometimes (50%) the 172.23.48.1 takes precedence above the the cluster DNS. This is undesirable because there's no DNS server listening on 172.23.48.1 or at least all kubernetes services are not resolvable.

Where can I configure docker/kubernetes to not use the DNS server other then the kubernetes one? 

```


Windows IP Configuration

   Host Name . . . . . . . . . . . . : d9c1f17fd931
   Primary Dns Suffix  . . . . . . . :
   Node Type . . . . . . . . . . . . : Hybrid
   IP Routing Enabled. . . . . . . . : No
   WINS Proxy Enabled. . . . . . . . : No
   DNS Suffix Search List. . . . . . : hnzbguap32ne3onlsis2rg1plh.ax.internal.cloudapp.net

Ethernet adapter vEthernet (Container NIC dff8d241):

   Connection-specific DNS Suffix  . :
   Description . . . . . . . . . . . : Hyper-V Virtual Ethernet Adapter #5
   Physical Address. . . . . . . . . : 00-15-5D-93-85-B1
   DHCP Enabled. . . . . . . . . . . : No
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : fe80::7c77:45af:355d:7c26%33(Preferred)
   IPv4 Address. . . . . . . . . . . : 10.244.5.172(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . :
   DNS Servers . . . . . . . . . . . : 10.3.0.10
   NetBIOS over Tcpip. . . . . . . . : Disabled

Ethernet adapter vEthernet (Container NIC d50ab1bd):

   Connection-specific DNS Suffix  . : hnzbguap32ne3onlsis2rg1plh.ax.internal.cloudapp.net
   Description . . . . . . . . . . . : Hyper-V Virtual Ethernet Adapter #6
   Physical Address. . . . . . . . . : 00-15-5D-F6-F2-89
   DHCP Enabled. . . . . . . . . . . : No
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : fe80::c004:6a45:619a:2086%37(Preferred)
   IPv4 Address. . . . . . . . . . . : 172.23.48.104(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.240.0
   Default Gateway . . . . . . . . . : 172.23.48.1
   DNS Servers . . . . . . . . . . . : 172.23.48.1
                                       10.3.0.10
   NetBIOS over Tcpip. . . . . . . . : Disabled
```

**Comment by colemickens** [comment]
cc: @JiangtianLi / @anhowe 

This is by design. The two NICs are intentional. One enables pod-to-pod network and the other serves as a gateway to the Internet.

**Comment by colemickens** [comment]
Where is this at @JiangtianLi ? Can we update this or close it, etc?

**Comment by skinny** [comment]
@colemickens I'm still having troubles with this. If there's something I can do to troubleshoot, please let me know.

**Comment by colemickens** [comment]
@JiangtianLi can you please provide an update

**Comment by colemickens** [comment]
cc: @anhowe

**Comment by JiangtianLi** [comment]
@skinny We are working on a solution to set the correct DNS server and make the order of NICs deterministic.

**Comment by JackQuincy** [comment]
@JiangtianLi has this work finished?

**Comment by JiangtianLi** [comment]
@JackQuincy Yes, the fix has gone into v1.5.3.

**Comment by JackQuincy** [comment]
closed as the fix has gone it.
